v1/lists.yaml

```diff
@@ -0,0 +1,1011 @@
+                token: '{{request.query.csrf_token}}'
+            return:
+              status: 200
+              headers:
+                content-type: application/json; charset=utf-8
+                cache-control: "no-cache"
+      x-monitor: false
+  /lists/teardown:
+    post:
+      tags:
+        - Reading lists
+      summary: Opt out from using reading lists.
+      description: |
+        Deletes all data. User needs to opt in again before being able to do anything.
+
+        Request must be authenticated with a MediaWiki session cookie.
+
+        Stability: [unstable](https://www.mediawiki.org/wiki/API_versioning#Unstable)
+      produces:
+        - application/json; charset=utf-8
+      parameters:
+        - name: csrf_token
+          in: query
+          required: true
+          schema: *csrf_token
+      responses:
+        '200':
+          description: Success.
+        default:
+          description: Error
+          schema:
+            $ref: '#/definitions/problem'
+      x-route-filters:
+        - path: ./lib/mediawiki_auth_filter.js
+      x-request-handler:
+        - forward_to_mw:
+            request:
+              uri: /{domain}/sys/action/rawquery
+              body:
+                action: readinglists
+                command: teardown
+                token: '{{request.query.csrf_token}}'
+            return:
+              status: 200
+              headers:
+                content-type: application/json; charset=utf-8
+                cache-control: "no-cache"
+      x-monitor: false
+  /lists/:
+    get:
+      tags:
+        - Reading lists
+      summary: Get all lists of the current user.
+      description: |
+        Returns metadata describing the lists of the current user. Might be truncated and include
+        a continuation token.
+
+        Request must be authenticated with a MediaWiki session cookie.
+
+        Stability: [unstable](https://www.mediawiki.org/wiki/API_versioning#Unstable)
+      parameters:
+        - name: next
+          in: query
+          description: Continuation parameter from previous request
+          type: string
+          required: false
+      produces:
+        - application/json; charset=utf-8; profile="https://www.mediawiki.org/wiki/Specs/Lists/0.1"
+      responses:
+        '200':
+          description: An array of list metadata.
+          schema:
+            type: object
+            properties:
+              lists:
+                type: array
+                items:
+                  $ref: '#/definitions/list_read'
+              next:
+                type: string
+                description: Continuation token.
+        default:
+          description: Error
+          schema:
+            $ref: '#/definitions/problem'
+      x-route-filters:
+        - path: ./lib/mediawiki_auth_filter.js
+      x-request-handler:
+        - forward_to_mw:
+            request:
+              uri: /{domain}/sys/action/rawquery
```

Comment:
Since this is an `action=query`, `/sys/action/query` should be used to enforce the response having a `query` field.

Reply:
As above, the response does not have any pages.